docs/03-Development-Process.md

```diff
@@ -12,13 +12,64 @@ examples :
 - feature/issue-15469
 - bugfix/issue-15469
 
+Origin vs Upstream in a Forked GitHub Workflow:
+
+| Term         | What it means                                               | Typical URL example                                | Used for                                         |
+| ------------ | ----------------------------------------------------------- | -------------------------------------------------- | ------------------------------------------------ |
+| **origin**   | Your fork of the repository (remote on your GitHub account) | `https://github.com/your-username/together.git`    | Pushing your rebased changes, creating PRs       |
+| **upstream** | The original repository you forked from                     | `https://github.com/Together-100Devs/together.git` | Pulling the latest changes from the main project |
+
 Developers should follow the following steps :
 
-- Updated the `main` : `git pull origin main`
-  - It is generally advised to keep your branch consistently up to date with main
-- create feature branch : `git checkout -b feature/issue-15469`
+- Create a feature branch : `git checkout -b feature/issue-15469`
 - Develop the specified requirements
-- commit and Push the code
+- Commit locally:
+
+  - stage your chages: `git add .`
+  - commit your changes: `git commit -m "your commit message"`
+
+- It is generally advised to keep your branch consistently up to date with `main`.
+
+  - Grab the latest upstream version of main, in case main had changes: `git fetch upstream`
+
+  - upstream = grabbing all upstream branches (upstream/main, upstream/add-dependabot, ect)
+
+- Rebase your feature branch on top of the latest upstream/main. We're telling git to use the "main" branch from upstream as the new "base" for your feature branch `git rebase upstream/main`
+
+- To keep the commit history clean, we require that you squash your commits into a single commit using interactive rebase:
+
+  - This shows all commits since you branched off of main: `git rebase -i upstream/main`
+
+  - Your editor will open up with a list of commits (oldest first). New to using the vim editor? Look below for a cheatsheet
+
+  > pick c32423434 first commit
+  > pick af243df second commit
+  > pick 234asf45 third commit
```

Comment:
```suggestion
  > pick c32423434 first commit  
  > pick af243df second commit  
  > pick 234asf45 third commit
```

Reply:
without the second space, everything was rendering as one line

<img width="863" height="111" alt="Image" src="https://github.com/user-attachments/assets/00b91274-bcad-46da-a257-8c04a820a8f3" />

Reply:
Good catch!